### DISCUSSION TITLE
Tab macro

### DISCUSSION THREAD

**Discussion by dada-harangus** [WikiTransformationProject/wikitraccs-releases#87] [Q&A]
In the documentation it says that the macro tabs are each one transformed to “normal” page content and all tabs are added one after another to the page.  But upon testing I only get this:
🚧👇 Replaced 'tabs-group' macro, its content follows below (disposition=='horizontal')

🚧👇 Replaced 'tab-pane' macro, its content follows below (anchor=='594760089' | name=='Overview')

🚧 Replaced Page Properties Report Macro (cql=='label = "competences" and space = currentSpace()' | headings=='Competence, C++SE, C++SSE, C++TS, C++TL, Capable Level, Strong Level, Expert Level')

🚧☝️ End of content of replaced 'tab-pane' macro

🚧👇 Replaced 'tab-pane' macro, its content follows below (anchor=='1942909752' | name=='Software Engineer')

🚧 Replaced Page Properties Report Macro (cql=='label = "competences" and space = currentSpace()' | headings=='Competence, C++SE, Capable Level, Strong Level, Expert Level')

🚧☝️ End of content of replaced 'tab-pane' macro

 

🚧👇 Replaced 'tab-pane' macro, its content follows below (anchor=='364836600' | name=='Senior Software Engineer')

🚧 Replaced Page Properties Report Macro (cql=='label = "competences" and space = currentSpace()' | headings=='Competence, C++SSE, Capable Level, Strong Level, Expert Level')

🚧☝️ End of content of replaced 'tab-pane' macro

 

🚧👇 Replaced 'tab-pane' macro, its content follows below (anchor=='2051781100' | name=='Technical Specialist')

🚧 Replaced Page Properties Report Macro (cql=='label = "competences" and space = currentSpace()' | headings=='Competence, C++TS, Capable Level, Strong Level, Expert Level')

🚧☝️ End of content of replaced 'tab-pane' macro

 

🚧👇 Replaced 'tab-pane' macro, its content follows below (name=='Technical Lead' | anchor=='1714856575')

🚧 Replaced Page Properties Report Macro (cql=='label = "competences" and space = currentSpace()' | headings=='Competence, C++TL, Capable Level, Strong Level, Expert Level')

🚧☝️ End of content of replaced 'tab-pane' macro

 Is something that I am doing wrong, or this is the expected behavior?

**Comment by heinrich-ulbricht** [comment]
Hm, looks like 5 tab panes that contained macros. This is to be expected. Each replaced tab pane is wrapped into a start and end marker, with the content in-between. Here the content is the page properties macro - which has no representation in SharePoint and is thus also replaced by placeholder.

Overall this looks expected. @dada-harangus What would your expectation be how this should be represented in SharePoint?

**Reply by dada-harangus** [reply]
I would expect the content from inside the tabs. In the documentation it says that the macro tabs are each one transformed to “normal” page content and all tabs are added one after another to the page.  I don`t care that it does not look like tabs anymore, but I do care about the missing content.

**Reply by heinrich-ulbricht** [reply]
@dada-harangus Is there any other content in those tabs apart from the **Page Properties Report Macro**? Your original posts suggests, that this macro is all that is in there.

Of course from a user perspective there is content. But for WikiTraccs it's "just" a macro that aggregates content from other places and displays it. This macro does not exist in SharePoint, so you get a placeholder.

I'm thinking about solving this, for this specific macro. There are two other macros, where WikiTraccs creates a "snapshot" of the macro content: the page tree and the attachments macro.

Would it make sense to migrate a static snapshot of what can be seen in Confluence? Note that when in Confluence something changes, e.g. page labels, the macro will update its content. This won't happen in SharePoint when using the "snapshot approach" as it will only be migrated as static text and links.

**Reply by dada-harangus** [reply]
For exemple I have a page like this: 
![exemple](https://github.com/WikiTransformationProject/wikitraccs-releases/assets/50258887/7ea3593d-b579-4796-9bdd-e6e8f923af33)
How can I migrate the text content regardless of how it looks in SharePoint?

**Reply by heinrich-ulbricht** [reply]
@dada-harangus To answer this I need to look under the hood of that page. Could you please send the page's storage format to contact @ wikitransformationproject.com -> Here's how to get it: [Get the Confluence Storage Format](https://www.wikitransformationproject.com/docs/troubleshooting/troubleshooting-strategies/get-confluence-storage-format)

You can redact the format as needed, but the more "original" it is, the better.
